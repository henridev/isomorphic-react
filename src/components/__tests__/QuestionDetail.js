import { mapStateToProps, QuestionDetailDisplay } from "../QuestionDetail";
import renderer, { create } from 'react-test-renderer'
import React from "react";

describe(`The Question Detail Component`, ()=>{
  describe('The container element', ()=>{
    describe('Map state to props', ()=> {
      it('should map state to props correctly', ()=>{
        const sampleQuestion = { 
          question_id: 1,
          title:'help me', 
          body:"what is the meaning of life?", 
          answer_count: 0
        }
        const appState = {
          questions: [
            sampleQuestion
          ]
        }
        const componentProps = {
          question_id: 1
        }
        const componentState = mapStateToProps(appState,componentProps)
        expect(componentState).toEqual(sampleQuestion)
      })
    })  
  })

  describe('The Display element', ()=>{
    it(`Should not regress`, ()=>{
      const tree = renderer.create(
        <QuestionDetailDisplay 
          question_id={1}
          title='help me!' 
          body="what is the meaning of life?"
          answer_count= {0}
          tags={['hitchhickers guide to the galaxy']}
        />
      )
      expect(tree.toJSON()).toMatchSnapshot()
    }) 
  })
  
})