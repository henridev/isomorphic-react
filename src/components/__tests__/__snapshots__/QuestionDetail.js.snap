// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Question Detail Component The Display element Should not regress 1`] = `
<div>
  <h3
    className="mb-2"
  >
    help me!
  </h3>
  <div>
    <div
      className="mb-3"
    >
      <div>
        <code>
          hitchhickers guide to the galaxy
          ?
        </code>
      </div>
    </div>
    <div>
      <p>
        what is the meaning of life?
      </p>
    </div>
    <div>
      0
       Answers
    </div>
  </div>
</div>
`;
